components/test_util/src/kv_generator.rs

```diff
@@ -11,24 +11,42 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-use rand::{self, Rng, ThreadRng};
+use rand::prelude::*;
+use rand::IsaacRng;
 
 /// A random generator of kv.
-/// Every iter should be taken in µs. See also `benches::bench_kv_iter`.
+///
+/// Every iteration should be taken in µs.
+#[derive(Clone, Debug)]
 pub struct KvGenerator {
     key_len: usize,
     value_len: usize,
-    rng: ThreadRng,
+    rng: IsaacRng,
 }
 
 impl KvGenerator {
     pub fn new(key_len: usize, value_len: usize) -> KvGenerator {
         KvGenerator {
             key_len,
             value_len,
-            rng: rand::thread_rng(),
+            rng: FromEntropy::from_entropy(),
         }
     }
+
+    pub fn with_seed(key_len: usize, value_len: usize, seed: u64) -> KvGenerator {
+        KvGenerator {
+            key_len,
+            value_len,
+            rng: IsaacRng::new_from_u64(seed),
+        }
+    }
+
+    /// Generate n pair of KVs.
+    ///
+    /// This function consumes current generator.
+    pub fn generate(self, n: usize) -> Vec<(Vec<u8>, Vec<u8>)> {
```

Comment:
How about keep ` generate_random_kvs()`? That is sample and useful, also I can add `generate_deliberate_kvs()` here.

Reply:
```rust
pub fn generate_deliberate_kvs(n: usize, key_len: usize, value_len: usize) -> Vec<(Vec<u8>, Vec<u8>)> {
    let kv_generator = KvGenerator::with_seed(key_len, value_len, 0);	
    kv_generator.take(n).collect()
}
```

Reply:
I think `KvGenerator::new(..).generate(..)` is more expressive and is not notably complex. Compared to the original `generate_random_kvs`, both of them only occupies 1 line with less than 50 characters. In addition, `generate_deliberate_kvs` is no more needed because it can be just written as `KvGenerator::from_seed(..).generate(..)`. What do you think?

Reply:
Either is fine.